File: algorithms/[1744]你能在你最喜欢的那天吃到你最喜欢的糖果吗？/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def canEat(self, candiesCount: List[int], queries: List[List[int]]) -> List[bool]:
        """
        前缀和
        time O(n + q), space(n), n 和 q 分别为 candiesCount 和 queries 长度
        """
        preSum = [0]
        for count in candiesCount:
            preSum.append(preSum[-1] + count)

        res: List[bool] = []
        for t, d, c in queries:
            min = 1 * (d + 1)
            max = c * (d + 1)
            res.append(not (min > preSum[t + 1] or max < preSum[t] + 1))

        return res

File: algorithms/[1744]你能在你最喜欢的那天吃到你最喜欢的糖果吗？/solution.ts
// topics = ["数组"]

/**
 * 前缀和
 * time O(n + q), space(n), n 和 q 分别为 candiesCount 和 queries 长度
 */
function canEat(candiesCount: number[], queries: number[][]): boolean[] {
  const preSum: number[] = [0];
  for (const count of candiesCount) {
    preSum.push(preSum[preSum.length - 1] + count);
  }

  return queries.map(([t, d, c]) => {
    const min = 1 * (d + 1);
    const max = c * (d + 1);
    return !(min > preSum[t + 1] || max < preSum[t] + 1);
  });
}
